src/Engine/Renderer/RenderTechnique/ShaderProgram.cpp

```diff
@@ -109,23 +100,35 @@ void ShaderProgram::loadShader( ShaderType type,
             { return a; }
         } );
 
-    auto fullsource = globjects::Shader::sourceFromString( shaderHeader + loadedSource->string() );
+    std::unique_ptr<globjects::StaticStringSource> fullsource{nullptr};
+    if ( fromFile )
+    {
+        LOG( logDEBUG ) << "Loading shader " << name;
+        auto loadedSource = globjects::Shader::sourceFromFile( name );
+        fullsource = globjects::Shader::sourceFromString( shaderHeader + loadedSource->string() );
+    }
+    else
+    { fullsource = globjects::Shader::sourceFromString( shaderHeader + name ); }
 
     // Radium V2 : allow to define global replacement per renderer, shader, rendertechnique ...
     auto shaderSource = globjects::Shader::applyGlobalReplacements( fullsource.get() );
 
-    // Workaround globject #include bug ...
-    // Radium V2 : update globject to see if tis bug is always here ...
+#if 1
+    // Workaround for #include directive
+    // Radium VB2 : rely on GL_ARB_shading_language_include to manage includes
     std::string preprocessedSource = preprocessIncludes( name, shaderSource->string(), 0 );
 
     auto ptrSource = globjects::Shader::sourceFromString( preprocessedSource );
-
-    addShaderFromSource( type, std::move( ptrSource ), name );
+#else
+    auto ptrSource = globjects::Shader::sourceFromString( shaderSource->string() );
```

Comment:
What's the purpose of this commented code?

Reply:
well, it's commented just two line up, it is the right way to do it, but not supported yet, so the active code is a workaround. I add a comment about that